voc/python/ast.py

```diff
@@ -617,8 +617,25 @@ def visit_Raise(self, node):
             for arg in args:
                 self.visit(arg)
 
-            self.context.add_opcodes(
-                java.Init(exception, *(['Lorg/python/Object;'] * len(args)))
+            for i in args:       #store the arguments passed in local variable #name
+                self.context.add_opcodes(
+                ASTORE_name(str(i)),
+                )
+
+            self.context.add_opcodes(    #declaring array of python objects 
+                ICONST_val(len(args)), 
+                JavaOpcodes.ANEWARRAY('org/python/types/Object'),
+            )
+            for i in args:      #storing the passed objects in the array
+                self.context.add_opcodes(  
+                  JavaOpcodes.DUP(),
+                  ICONST_val(len(args)-args.index(i)-1),
+                  ALOAD_name(str(i)),
+                 JavaOpcodes.AASTORE(),
+                )
+
+            self.context.add_opcodes(     
+                java.Init(exception, '[Lorg/python/Object;')
```

Comment:
This looks like it would work, but instead of creating variable names, it will be better to update the `Array()` helper class in `voc/python/types/java.py` to support creating arrays loading the values from the stack.

That class supports an argument `fill` that is used to fill the array.
We can make it a special value `'STACK'` that would load the values from the stack instead.

Here is what the code would look like:

```
        if self.fill:
            if self.fill == 'STACK':
                for i in range(self.size):
                    context.add_opcodes(
                         # the next two instructions modify the stack like: 1 2 -> 2 2 1
                        JavaOpcodes.DUP_X1(),
                        JavaOpcodes.SWAP(),

                        ICONST_val(i),
                        JavaOpcodes.SWAP(),
                        JavaOpcodes.AASTORE(),
                    )
            else:
                for i in range(self.size):
                    context.add_opcodes(
                        JavaOpcodes.DUP(),
                        ICONST_val(i),
                        self.fill,
                        JavaOpcodes.AASTORE(),
                    )
```

Then we can use this function to build the array to pass to `java.Init`.

Does this make sense?

Feel free to ask any questions, and thanks for getting this far! :)